RecoLocalCalo/HGCalRecProducers/python/HGCalUncalibRecHit_cfi.py

```diff
@@ -47,3 +47,10 @@
 
     algo = cms.string("HGCalUncalibRecHitWorkerWeights")
 )
+
+from Configuration.ProcessModifiers.premix_stage2_cff import premix_stage2
+premix_stage2.toModify(HGCalUncalibRecHit,
+    HGCEEdigiCollection = 'mixData:HGCDigisEE',
```

Comment:
is it possible to have a unique InputTag (e.g. hgcDigisEE) by appropriately setting an EDAlias?
It would be a much better solution, instead of the one that requires special modifications for every consumer module

Reply:
Not easily. The original InputTag here is `mix:HGCDigisEE`, where `mix` stands for the `MixingModule` for the signal event which is part of the same job where `mixData` is run. So it is not possible to define `EDAlias` for `mix` without renaming `mix` first to something else (at which point I'd rename `mixData` to `mix`) and dealing with all the knock-on effects of that (which, on the other hand, should be confined in the `DIGI` step as ideally nothing signal-only should "leak" to the later steps, so maybe such an approach could be explored, but that goes well beyond this PR).

On a general note, I agree on the maintenance argument, but personally I find already the existing (pre)mixing EDAliases
https://github.com/cms-sw/cmssw/blob/master/SimGeneral/MixingModule/python/aliases_PreMix_cfi.py
https://github.com/cms-sw/cmssw/blob/master/SimGeneral/MixingModule/python/aliases_cfi.py
confusing when trying to understand / debug on what exactly is happening in (pre)mixing, especially because they can't be defined universally because of module name clashes. I feel 
the extra layer of indirection adds mental load.

Reply:
Premix stage2 job indeed runs both `mix` (digitization for signal) and `mixData` (overlay pileup on top of signal).

In a sense the current premixing infrastructure has been "tuned" to use the standard module names for the signal "mixing", and the premixing output needs special treatment. For most cases the premixing output gets packed to RAW, so all those will look as normal for clients. It is mainly the simulation-level stuff (like simDigis as in this case, and MC truth) needing a special treatment.

One could think of changing the system to always preserve the labels of products that are stored in the output of DIGI step. The price is then something along
* customize the module labels of the DIGI step in case of premixing (not easy), or
* add additional layer of EDAliases, leading to changing many of the labels of the current DIGI products

All this is definitively for longer term. Maybe we should open an issue to not forget?

(And to clarify, also to myself, my earlier dislike of EDAlias: most of my dislike of the current system is that some things go through EDAlias and some things do not, and these sets are different between classical mixing and premixing. Being consistent would make the setup easier to understand.)